src/Wallabag/ImportBundle/Import/BrowserImport.php

```diff
@@ -0,0 +1,227 @@
+
+            return false;
+        }
+
+        $data = json_decode(file_get_contents($this->filepath), true);
+
+        if (empty($data)) {
+            return false;
+        }
+
+        $this->nbEntries = 1;
+        $this->parseEntries($data);
+        $this->em->flush();
+
+        return true;
+    }
+
+    private function parseEntries($data)
+    {
+        foreach ($data as $importedEntry) {
+            $this->parseEntry($importedEntry);
+        }
+        $this->totalEntries += count($data);
+    }
+
+    private function parseEntry($importedEntry)
+    {
+        if (!is_array($importedEntry)) {
+            return;
+        }
+
+        /* Firefox uses guid while Chrome uses id */
+
+        if ((!key_exists('guid', $importedEntry) || (!key_exists('id', $importedEntry))) && is_array(reset($importedEntry))) {
+            $this->parseEntries($importedEntry);
+
+            return;
+        }
+        if (key_exists('children', $importedEntry)) {
+            $this->parseEntries($importedEntry['children']);
+
+            return;
+        }
+        if (key_exists('uri', $importedEntry) || key_exists('url', $importedEntry)) {
+
+            /* Firefox uses uri while Chrome uses url */
+
+            $firefox = key_exists('uri', $importedEntry);
+
+            $existingEntry = $this->em
+                ->getRepository('WallabagCoreBundle:Entry')
+                ->findByUrlAndUserId(($firefox) ? $importedEntry['uri'] : $importedEntry['url'], $this->user->getId());
+
+            if (false !== $existingEntry) {
+                ++$this->skippedEntries;
+
+                return;
+            }
+
+            if (false === parse_url(($firefox) ? $importedEntry['uri'] : $importedEntry['url']) || false === filter_var(($firefox) ? $importedEntry['uri'] : $importedEntry['url'], FILTER_VALIDATE_URL)) {
+                $this->logger->warning('Imported URL '.($firefox) ? $importedEntry['uri'] : $importedEntry['url'].' is not valid');
+                ++$this->skippedEntries;
+
+                return;
+            }
+
+            try {
+                $entry = $this->contentProxy->updateEntry(
+                    new Entry($this->user),
+                    ($firefox) ? $importedEntry['uri'] : $importedEntry['url']
+                );
+            } catch (\Exception $e) {
+                $this->logger->warning('Error while saving '.($firefox) ? $importedEntry['uri'] : $importedEntry['url']);
+                ++$this->skippedEntries;
+
+                return;
+            }
+
+            $entry->setArchived($this->markAsRead);
+
+            $this->em->persist($entry);
+            ++$this->importedEntries;
+
+            // flush every 20 entries
+            if (($this->nbEntries % 20) === 0) {
+                $this->em->flush();
+                $this->em->clear($entry);
+            }
+            ++$this->nbEntries;
+
+            /*
+
+            Maybe not useful. Delete at will.
```

Comment:
I had tried http://stackoverflow.com/a/9853018 but it was not satisfying. Forgot to delete comment too it seems.